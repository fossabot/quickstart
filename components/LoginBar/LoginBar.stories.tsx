import { Meta, Story } from '@storybook/react'
import React from 'react'

import { LoginBar } from './LoginBar'

const config: Meta = {
  title: 'Components/LoginBar',
  component: LoginBar,
}

export default config

const Template: Story = (args) => <LoginBar {...args} />

export const Default = Template.bind({})
Default.parameters = {
  jest: ['LoginBar.test.tsx'],
}

export const LoggedIn = Template.bind({})
LoggedIn.args = {}

export const LoggedOut = Template.bind({})
LoggedOut.args = {}
