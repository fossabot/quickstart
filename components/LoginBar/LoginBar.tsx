import { useSession } from 'next-auth/client'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'

import { PAGES } from '../../interface/auth/constants'
import ErrorBoundary from '../../util/ErrorBoundary'
import { Button } from '../Button/Button'

export function LoginBar(): JSX.Element {
  const [session, loading] = useSession()
  const hasSession = session != null

  const { t } = useTranslation()

  return (
    <ErrorBoundary>
      <div className="flex items-center justify-end space-x-2">
        {hasSession ? (
          <Button
            data-test-id="SignOutButton"
            href={PAGES.signOut}
            label={t('common:signOut')}
          />
        ) : (
          <>
            <Button
              data-test-id="SignInButton"
              disabled={loading}
              href={PAGES.signIn}
              size="small"
              label={t('common:signIn')}
            />
            <Button
              data-test-id="SignUpButton"
              disabled={loading}
              href={PAGES.signIn}
              primary
              label={t('common:signUp')}
            />
          </>
        )}
      </div>
    </ErrorBoundary>
  )
}
