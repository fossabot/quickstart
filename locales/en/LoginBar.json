{
  "signIn": "Sign In",
  "signOut": "Sign Out",
  "signUp": "Sign Up"
}
