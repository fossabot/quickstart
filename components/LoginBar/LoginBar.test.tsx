import React from 'react'

import { LoginBar } from './LoginBar'

describe('LoginBar', () => {
  it('Should render without an error', () =>
    expect(() => <LoginBar />).not.toThrow())

  it('Should match the snapshot', () => expect(<LoginBar />).toMatchSnapshot())
})
