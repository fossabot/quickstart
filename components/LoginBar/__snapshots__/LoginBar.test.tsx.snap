// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit/Components/LoginBar Should match the snapshot 1`] = `<LoginBar />`;
